### PR TITLE
Check average_rate availability in video reader

### PR DESCRIPTION
**Pull request by stefan-baumann**
This PR addresses #8420 by adding a simple check of whether the `average_rate` attribute exists on the stream in question before trying to access it. The general behavior is not changed by this PR, as the original fallback to using `stream.sample_rate` instead is retained.

### CODE REVIEW

File: torchvision/io/video_reader.py [hunk 1]
```diff
@@ -251,7 +251,12 @@ def get_metadata(self) -> Dict[str, Any]:
                         rate_n = "framerate"
                     metadata[stream.type] = {rate_n: [], "duration": []}
 
-                rate = stream.average_rate if stream.average_rate is not None else stream.sample_rate
+                    rate = (
+                        stream.average_rate
+                        if hasattr(stream, "average_rate")
+                        and stream.average_rate is not None
+                        else stream.sample_rate
+                    )
```
Comment (NicolasHug) [inline torchvision/io/video_reader.py:259]: Thanks for the PR, I think this should be equivalent to

```suggestion
                    rate = getattr(stream, "average_rate", None) or stream.sample_rate
```

Comment (stefan-baumann) [reply torchvision/io/video_reader.py]: Yes, thanks, that's even better!

File: torchvision/io/video_reader.py [hunk 2]
```diff
@@ -251,7 +251,7 @@ def get_metadata(self) -> Dict[str, Any]:
                         rate_n = "framerate"
                     metadata[stream.type] = {rate_n: [], "duration": []}
 
-                rate = stream.average_rate if stream.average_rate is not None else stream.sample_rate
+                    rate = getattr(stream, "average_rate", None) or stream.sample_rate
```
Comment (NicolasHug) [inline torchvision/io/video_reader.py:254]:
```suggestion
                rate = getattr(stream, "average_rate", None) or stream.sample_rate
```

**Comment by NicolasHug** [comment]
Thanks a lot for the fix @stefan-baumann

**Pull request merged**